### DISCUSSION TITLE
PCL 国际化翻译的翻译人员参与方式探讨 (Exploration of Translation Participation Methods for PCL Internationalization)

### DISCUSSION THREAD

**Discussion by tangge233** [Meloong-Git/PCL#4262] [讨论]
#4145 的合并让 PCL 基本上支持了国际化，然而临时上传的机翻以及往后更多的功能的语言翻译支持都需要翻译人员的参与。此 Discussion 正是讨论一些可行的让翻译人员参加翻译的方式。
The merge of #4145 has essentially enabled PCL to support internationalization, however, the interim use of machine translations and future support for language translations of more features will require the involvement of translators. This Discussion aims to explore some feasible ways for translators to participate in the translation process.

截至发布时一些已有的翻译参加方式想法：(As of the time of publication, some existing ideas for translator participation include:)
[@Silverteal 云翻译平台](https://github.com/Hex-Dragon/PCL2/pull/4145#issuecomment-2212421901)
[@Pigeon0v0 翻译文本临时挂载](https://github.com/Hex-Dragon/PCL2/pull/4145#issuecomment-2227627728)

欢迎各位的积极讨论
We welcome everyone's active discussion and input.

**Comment by Pigeon0v0** [comment]
翻译文本挂载是为了方便本地测试，云端平台不方便测试...

**Reply by tangge233** [reply]
之后再开个 PR 啦……

**Reply by Silverteal** [reply]
合理配置的云端平台不应该存在这种问题吧

**Comment by Silverteal** [comment]
事实上即使是成熟的项目，也会存在翻译人员翻完了发现显示区域放不下的情况，这种情况下一般是有人发现后再返工的。也有些开发者会先用一些乱打的文字测试出每个字符串最长的长度（称为“伪本地化语言”），并在开发者提示中指定长度要求。

对于PCL这种设计之初并没有充分考虑到国际化需求的项目来说，进行伪本地化或许会比较有效率。

**Comment by Chiloven945** [comment]
目前来说能进行翻译的方式应该就只有云翻译平台和提交pr，除此之外也别无他法了。

**Reply by tangge233** [reply]
Crowdin 的免费计划放得下 PCL 的翻译吗？

**Comment by Silverteal** [comment]
很紧张，多加几个语言就会超限




------------------&nbsp;Original&nbsp;------------------
From:                                                                                                                        "Hex-Dragon/PCL2"                                                                                    ***@***.***&gt;;
Date:&nbsp;Fri, Aug 9, 2024 09:50 AM
***@***.***&gt;;
***@***.******@***.******@***.***&gt;;
Subject:&nbsp;Re: [Hex-Dragon/PCL2] PCL 国际化翻译的翻译人员参与方式探讨 (Exploration of Translation Participation Methods for PCL Internationalization) (Discussion #4262)





 
Crowdin 的免费计划放得下 PCL 的翻译吗？

**Reply by tangge233** [reply]
先别加那么多语言……先把已有的4种弄好再说……

**Comment by Silverteal** [comment]
你这话说的，后面换过来不是更麻烦？沉没成本也要考虑的




------------------&nbsp;Original&nbsp;------------------
From:                                                                                                                        "Hex-Dragon/PCL2"                                                                                    ***@***.***&gt;;
Date:&nbsp;Fri, Aug 9, 2024 11:09 AM
***@***.***&gt;;
***@***.******@***.******@***.***&gt;;
Subject:&nbsp;Re: [Hex-Dragon/PCL2] PCL 国际化翻译的翻译人员参与方式探讨 (Exploration of Translation Participation Methods for PCL Internationalization) (Discussion #4262)





 
先别加那么多语言……先把已有的4种弄好再说……

**Reply by Chiloven945** [reply]
基本上所有翻译平台都可以导出XLIFF……
